packages/commonwealth/client/scripts/views/components/sidebar/index.tsx

```diff
@@ -19,99 +16,100 @@ import { GovernanceSection } from './governance_section';
 import { SidebarQuickSwitcher } from './sidebar_quick_switcher';
 import { CWIcon } from '../component_kit/cw_icons/cw_icon';
 import { CWText } from '../component_kit/cw_text';
-import withRouter from '../../../navigation/helpers';
+import { useCommonNavigate } from '../../../navigation/helpers';
+import useForceRerender from 'hooks/useForceRerender';
 
 export type SidebarMenuName =
   | 'default'
   | 'createContent'
   | 'exploreCommunities';
 
-class SidebarComponent extends ClassComponent {
-  view() {
-    const activeAddressRoles = app.roles.getAllRolesInCommunity({
-      chain: app.activeChainId(),
-    });
+export const Sidebar = () => {
+  const navigate = useCommonNavigate();
+  const forceRerender = useForceRerender();
 
-    app.sidebarRedraw.on('redraw', () => {
-      this.redraw();
-    });
+  const activeAddressRoles = app.roles.getAllRolesInCommunity({
+    chain: app.activeChainId(),
+  });
 
-    const currentChainInfo = app.chain?.meta;
+  useEffect(() => {
+    app.sidebarRedraw.on('redraw', () => forceRerender());
 
-    const onHomeRoute =
-      _DEPRECATED_getRoute() === `/${app.activeChainId()}/feed`;
+    return () => {
+      app.sidebarRedraw.on('redraw', () => forceRerender());
```

Comment:
it does not do anything. On return it should remove listener so something like removeListener() / off()

Reply:
Deleted the code and nothing seems to break, so 🤷 .

Reply:
Here there is nothing to break down, here it is a matter of good practice. If you activate the listner on you component mount, you should deactive it on unmount. Otherwise, in some cases it might lead to memory leaks.

Reply:
Alternatively you can use `once()` if it's a one-off redraw -- that will automatically unmount the listener.